src/toil/test/batchSystems/batchSystemTest.py

```diff
@@ -341,10 +362,33 @@ def testJobConcurrency(self):
                 for _ in range(self.cpuCount):
                     root.addFollowOn(Job.wrapFn(measureConcurrency, counterPath, self.sleepTime,
                                                 cores=coresPerJob, memory='1M', disk='1Mi'))
-                Job.Runner.startToil(root, options)
+                with Toil(options) as toil:
+                    toil.start(root)
                 _, maxValue = getCounters(counterPath)
                 self.assertEqual(maxValue, self.cpuCount // coresPerJob)
 
+        def test_omp_threads(self):
+            """
+            Test if the OMP_NUM_THREADS env var is set correctly based on jobs.cores.
+            """
+            test_cases = {
+                # mapping of the number of cores to the OMP_NUM_THREADS value
+                0.1: "1",
+                1: "1",
+                2: "2"
+            }
+
+            temp_dir = self._createTempDir()
+            options = self.getOptions(temp_dir)
+
+            for cores, expected_omp_threads in test_cases.items():
+                if os.environ.get('OMP_NUM_THREADS'):
+                    expected_omp_threads = os.environ.get('OMP_NUM_THREADS')
+                    logger.info(f"OMP_NUM_THREADS is set.  Using OMP_NUM_THREADS={expected_omp_threads} instead.")
```

Comment:
When is this branch ever going to happen?

Reply:
This is not going to happen on the CI but I was testing with it locally for the `single_machine` batch system.